### ISSUE TITLE
Allow for configurable Kerberos Acceptor name

### ISSUE THREAD

**Issue by Thulium-Drake** [cockpit-project/cockpit#19926]
Page: 

Hi there!

I'd really like to be able to use the HTTP service principal for the actual webserver on a system and not need to share the keytab between the services. Would it be possible to make the Kerberos Acceptor name configurable, such as Apache does as well?  (https://github.com/gssapi/mod_auth_gssapi?tab=readme-ov-file#gssapiacceptorname)

I'd suggest a default (if you wish to change it) of COCKPIT/host.example.com@EXAMPLE.COM

Thanks!

**Comment by martinpitt** [comment]
Technically you can -- this is defined in [cockpit-certificate-helper](https://github.com/cockpit-project/cockpit/blob/main/src/tls/cockpit-certificate-helper.in), see `cmd_ipa_request()`. I.e. if you create /etc/cockpit/krb5.config yourself, then it'll use that principal.

**Comment by Thulium-Drake** [comment]
Thanks for the reply! I took a look at the code you mentioned:

https://github.com/cockpit-project/cockpit/blob/69633839c8c26d1fbeac010b6634d1fd373f652c/src/tls/cockpit-certificate-helper.in#L129

Isn't the kerberos principal hardcoded here? I don't see a reference to the config file you mention in this file. I can see it will handle all interactions with the IPA server in order to set up the principal and certificate. But I don't see a means to configure (or overwrite) the service name used.

**Comment by martinpitt** [comment]
> Isn't the kerberos principal hardcoded here?

Yes, but [there is a check](https://github.com/cockpit-project/cockpit/blob/main/src/tls/cockpit-certificate-helper.in#L103) which won't create the keytab if it's already present.

> I don't see a reference to the config file you mention in this file. 

It's a bit indirect:
```sh
COCKPIT_CONFIG="@sysconfdir@/cockpit"
local KEYTAB="${COCKPIT_CONFIG}/krb5.keytab"
```
and `@sysconfdir@` gets filled in by autoconf from  `./configure --sysconfdir`, i.e. pretty much always `/etc` in standard distro packages.
